common/elasticsearch/client/v6/client.go

```diff
@@ -0,0 +1,205 @@
+		elastic.SetDecoder(&elastic.NumberDecoder{}), // critical to ensure decode of int64 won't lose precise)
+	}
+	if connectConfig.DisableSniff {
+		clientOptFuncs = append(clientOptFuncs, elastic.SetSniff(false))
+	}
+	if connectConfig.DisableHealthCheck {
+		clientOptFuncs = append(clientOptFuncs, elastic.SetHealthcheck(false))
+	}
+
+	if awsSigningClient != nil {
+		clientOptFuncs = append(clientOptFuncs, elastic.SetHttpClient(awsSigningClient))
+	}
+
+	if tlsClient != nil {
+		clientOptFuncs = append(clientOptFuncs, elastic.SetHttpClient(tlsClient))
+	}
+
+	client, err := elastic.NewClient(clientOptFuncs...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ElasticV6{
+		client: client,
+		logger: logger,
+	}, nil
+}
+
+// Refer to https://github.com/olivere/elastic/blob/release-branch.v6/recipes/aws-connect-v4/main.go
+func buildSigningHTTPClientFromStaticCredentialV6(credentialConfig config.AWSStaticCredential) (*http.Client, error) {
+	awsCredentials := credentials.NewStaticCredentials(
+		credentialConfig.AccessKey,
+		credentialConfig.SecretKey,
+		credentialConfig.SessionToken,
+	)
+	return esaws.NewV4SigningClient(awsCredentials, credentialConfig.Region), nil
+}
+
+func buildSigningHTTPClientFromEnvironmentCredentialV6(credentialConfig config.AWSEnvironmentCredential) (*http.Client, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(credentialConfig.Region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return esaws.NewV4SigningClient(sess.Config.Credentials, credentialConfig.Region), nil
+}
+
+func (c *ElasticV6) PutMapping(ctx context.Context, index, body string) error {
+	_, err := c.client.PutMapping().Index(index).Type("_doc").BodyString(body).Do(ctx)
+	return err
+}
+
+func (c *ElasticV6) CreateIndex(ctx context.Context, index string) error {
+	_, err := c.client.CreateIndex(index).Do(ctx)
+	return err
+}
+
+func (c *ElasticV6) Count(ctx context.Context, index, query string) (int64, error) {
+	return c.client.Count(index).BodyString(query).Do(ctx)
+}
+
+func (c *ElasticV6) ClearScroll(ctx context.Context, scrollID string) error {
+	return elastic.NewScrollService(c.client).ScrollId(scrollID).Clear(ctx)
+}
+func (c *ElasticV6) Scroll(ctx context.Context, index, body, scrollID string) (*client.Response, error) {
+	scrollService := elastic.NewScrollService(c.client)
+	var esResult *elastic.SearchResult
+	var err error
+
+	// we are not returning error immediately here, as result + error combination is possible
+	if len(scrollID) == 0 {
+		esResult, err = scrollService.Index(index).Body(body).Do(ctx)
+	} else {
+		esResult, err = scrollService.ScrollId(scrollID).Do(ctx)
+	}
+
+	var byteResult [][]byte
+	if esResult != nil && esResult.Hits != nil {
+		for _, hit := range esResult.Hits.Hits {
+			byteResult = append(byteResult, *hit.Source)
+		}
+	}
+
+	result := &client.Response{
+		TookInMillis: esResult.TookInMillis,
```

Comment:
panic if `esResult` is `nil`. We should do a check on `esResult` and `err` and return early.

Reply:
👍 thanks!